SPEC.md

```diff
@@ -229,10 +242,15 @@ Retrieve operation result.
 
 Retrieve operation details.
 
-**Path**: `/{service}/{operation}/{operation_id}`
```

Comment:
I do not think we should remove the token from the URL path on these. At the very least it should be a query parameter, though in the URL path makes more sense IMO. Using headers as required parameters to specific HTTP calls is strange (as opposed to general metadata like auth).

Reply:
This is similar to the callback token where we decided to not put it in the URL since URLs are typically logged in access logs and we don't want potentially sensitive information leaking there.

Reply:
If it is sensitive shouldn't it be encrypted?

Reply:
Is token meant to be more sensitive than ID for this use case? I calling a string to reference the operation by an "id" or "token" doesn't really change the purpose. Requiring headers for these calls to work is rough devexp.

Reply:
> If it is sensitive shouldn't it be encrypted?

Yes, but it still doesn't stop someone from potentially stealing your token.

Reply:
Or your ID from before or whatever else we choose to call the single string to reference an operation. Why does the English word we use to define this string affect whether it's allowed in the URL?

Reply:
The original designation was a single business ID, we now want to expand this to be any arbitrary token. This token could potentially contain an OTP for retrieving the result.

Reply:
It always could have contained that

Reply:
I agree that's generally true for RESTful APIs, but I don't think that's true for tokens. Sure, it could have always technically contained that, but we didn't have that in mind in the original design.

Reply:
My concern is this is a significant usability regression of the Nexus HTTP API. Is it possible to not go backwards in dev experience? For instance, maybe this header approach can be an alternative instead of the required way to give this parameter?

Reply:
I was just thinking the same thing. We could have two approaches, so users that are concerned with tokens leaking could send it as a header. That's essentially what the related Go SDK PR allows.

Let's keep it path based for now and rename, we can add the header approach later.

Reply:
Working on this now and feeling a bit on the fence here. Let me think about this some more.

Reply:
Pushed a change, LMK what you think.

Reply:
LGTM. Could also understand if we wanted to reserve some paths for an eventual no-token future, or I could also understand if you wanted token to be a query parameter instead of path piece so it can be more easily made optional in lieu of a header later.